pkg/operator/sync.go

```diff
@@ -52,9 +52,12 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 	}
 	glog.V(3).Info("Synced up all machine-api-controller components")
 
-	// In addition, if the Provider is BareMetal, then bring up
-	// the baremetal-operator pod
+	// In addition, if the Provider is BareMetal, then imay have to deploy metal3 pod.
 	if config.BaremetalControllers.BaremetalOperator != "" {
+		if !isBaremetalProvisioningConfigAvailable(optr.dynamicClient, baremetalProvisioningCR) {
+			glog.V(3).Info("Not necessary to deploy metal3 pod for this baremetal install.")
```

Comment:
This could be misleading if there are transient connection issues, perhaps this should be reworded to say something more similar to
```suggestion
			glog.V(3).Info("Unable to read Baremetal Provisioning config, skipping Metal3 install.")
```

Reply:
Refactored code so do not need this Info message. I think with the current refactor, even if there are transient connection issues, the sync operation would re-try deployment and eventually succeed but without these messages in the log.